### PR TITLE
feat: circleci matrix job

### PR DESCRIPTION
**Pull request by gerasyana**
Upgraded CircleCI configs to run **checkout_and_test** job to for both mongo 3.6 and mongo 4.4 versions. Use cached build from `mongo4_4` executor in `publish` and `publish_docs` jobs.

- [x] Add two executors
- [x] Update **save_cache** step

https://circleci.com/blog/circleci-matrix-jobs/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+version: 2.1
+
+executors:
+  mongo3_6:
     docker:
       - image: circleci/node:10
       - image: circleci/mongo:3.6-jessie
+  mongo4_4:
+    docker:
+      - image: circleci/node:10
```
Comment (thekiwi) [inline .circleci/config.yml:10]: can we also update all the node versions to 14? here it's `circleci/node:10` and further down it's `circleci/node:8-stretch`

**Review by thekiwi — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (gerasyana) [reply .circleci/config.yml:10]: Yes I can
What about running jobs for both of 10 and 14 versions , but use image built with mongo 4.4 and node 14 with for publishing?

Comment (thekiwi) [reply .circleci/config.yml:10]: Sounds good, though I think if we're doing multiple Node builds, then we should do 10, 12 and 14?

Comment (gerasyana) [reply .circleci/config.yml:10]: Yes we can. In this case there will be 6 jobs

Comment (gerasyana) [reply .circleci/config.yml]: How **build_and_publish** workflow looks like now. I also added a conditional step **save_cache** for the job with **mongo:4.4** and  **node:14** docker images. From my view it doesn't make sense to cache images for each of the jobs.

Screen Shot 2021-01-26 at 12 18 32

**Review by pierissimo — APPROVED** [review]
very nice 👍

**Review by thekiwi — APPROVED** [review]
Amazing! 👏🏻

**Pull request merged**